utildb/gen_deleted_accounts_test.go

```diff
@@ -0,0 +1,242 @@
+	encounteredErrors := errorHandler(stopChan, errChan)
+	errChan <- fmt.Errorf("error1")
+	errChan <- fmt.Errorf("error2")
+	close(errChan)
+	got := <-encounteredErrors
+	assert.Equal(t, "error1\nerror2", got.Error())
+}
+
+func Test_errorHandlerResultGetsClosed(t *testing.T) {
+	stopChan := make(chan struct{})
+	errChan := make(chan error, 2)
+	encounteredErrors := errorHandler(stopChan, errChan)
+	close(errChan)
+	err, ok := <-encounteredErrors
+	if !ok {
+		t.Errorf("encounteredErrors channel should be open with result")
+	}
+	if err != nil {
+		t.Errorf("encounteredErrors shouldn't have any errors")
+	}
+
+	_, ok = <-encounteredErrors
+	if ok {
+		t.Errorf("encounteredErrors channel should be closed")
+	}
+}
+
+func Test_launchWorkersParallel(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	processor := executor.NewMockTxProcessor(ctrl)
+
+	wg := &sync.WaitGroup{}
+	cfg := &utils.Config{Workers: 3, ChainID: 250}
+
+	stopChan := make(chan struct{})
+	errChan := make(chan error)
+
+	testTx := makeTestTx(6)
+
+	// channel for each worker to get tasks for processing
+	workerInputChannels := make(map[int]chan *substate.Transaction)
+	for i := 0; i < cfg.Workers; i++ {
+		workerInputChannels[i] = make(chan *substate.Transaction)
+		go func(workerId int, workerIn chan *substate.Transaction) {
+			// 2 transactions for each worker
+			workerIn <- &substate.Transaction{Block: uint64(workerId), Substate: testTx[workerId]}
+			workerIn <- &substate.Transaction{Block: uint64(workerId + cfg.Workers), Substate: testTx[workerId+cfg.Workers]}
+			close(workerIn)
+		}(i, workerInputChannels[i])
+	}
+
+	processor.EXPECT().ProcessTransaction(gomock.Any(), 0, gomock.Any(), substatecontext.NewTxContext(testTx[0])).Return(nil, nil)
+	processor.EXPECT().ProcessTransaction(gomock.Any(), 1, gomock.Any(), substatecontext.NewTxContext(testTx[1])).Return(nil, nil)
+	processor.EXPECT().ProcessTransaction(gomock.Any(), 2, gomock.Any(), substatecontext.NewTxContext(testTx[2])).Return(nil, nil)
+	processor.EXPECT().ProcessTransaction(gomock.Any(), 3, gomock.Any(), substatecontext.NewTxContext(testTx[3])).Return(nil, nil)
+	processor.EXPECT().ProcessTransaction(gomock.Any(), 4, gomock.Any(), substatecontext.NewTxContext(testTx[4])).Return(nil, nil)
+	processor.EXPECT().ProcessTransaction(gomock.Any(), 5, gomock.Any(), substatecontext.NewTxContext(testTx[5])).Return(nil, nil)
+
+	outPut := launchWorkers(wg, cfg, workerInputChannels, processor, stopChan, errChan, nil)
+
+	var orderCheck uint64 = 0
+	// 2 transactions for each worker
+	for k := 0; k < 2; k++ {
+		for i := 0; i < cfg.Workers; i++ {
+			r, ok := <-outPut[i]
+			if !ok {
+				t.Fatalf("results channel should be open")
+			}
+			if orderCheck != r.tx.Block {
+				t.Fatalf("results are in incorrect order")
+			}
+			orderCheck++
+		}
+	}
+	wg.Wait()
+
+	ctrl.Finish()
+}
+
+// makeTestTx creates dummy substates that will be processed without crashing.
+func makeTestTx(count int) []*substate.Substate {
```

Comment:
Can you specify what does the test suppose to test please

Reply:
see header comment